strider/server.py

```diff
@@ -359,13 +359,23 @@ async def extract_results(query_id, since, limit, offset, database):
     ]
 
 
-@APP.post('/plan', response_model=Dict)
+@APP.post('/plan', response_model=List[Dict])
 async def generate_traversal_plan(
         query: Query,
-) -> list[Dict]:
+) -> List[Dict]:
     """Generate plans for traversing knowledge providers."""
     query_graph = query.message.query_graph.dict()
-    return await generate_plans(query_graph)
+    plans = await generate_plans(query_graph)
+
+    plans_stringified_keys = []
+    for plan in plans:
+        plans_stringified_keys.append(
+            {
+                f"{step.source}-{step.edge}->{step.target}": v
```

Comment:
Doesn't the arrow direction depend on the query graph?

Reply:
In this case the arrow indicates the source and target of the edge in the query graph. The predicate direction is indicated in the plan itself. I can remove the arrow and just have n0-e01-n1 if that would be less confusing.

Reply:
Our convention is that the operation direction is indicated by the order of the nodes on the line and in most (all?) contexts, an arrow indicates predicate direction, so I think leaving it off here is a good choice.